### PR TITLE
Initial scripts for building Parcel and CSD

### PR DESCRIPTION
**Pull request by wangmiao1002**
Kyuubi parcel and csd script

### CODE REVIEW

**Comment by pan3793** [comment]
Thanks, I will take a look today.

File: bin/kyuubibulid.sh [hunk 1]
```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+set -x
+set -e
+#cm_ext branch
+CM_EXT_BRANCH=cm5-5.15.0
+kyuubi_version=1.6.1
+parcel_version=${kyuubi_version}-1
+kyuubi_package_url=https://archive.apache.org/dist/incubator/kyuubi/kyuubi-${kyuubi_version}-incubating/apache-kyuubi-${kyuubi_version}-incubating-bin.tgz
```
Comment (pan3793) [inline bin/kyuubibulid.sh]: `-incubating` should be part of version, and since Kyuubi 1.7.0 has been released, we can move to 1.7.0 instead.

Comment (wangmiao1002) [reply bin/kyuubibulid.sh]: Ok, thank you for your reply. Please check again

**Comment by pan3793** [comment]
cc @bowenliang123 as well

File: parcel/kyuubi-parcel-src/meta/alternatives.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+    "kbeeline": {
+      "destination": "/usr/bin/kbeeline",
```
Comment (pan3793) [inline parcel/kyuubi-parcel-src/meta/alternatives.json]: To distinguish w/ hive beeline, maybe we should rename our scripts to `kyuubi-beeline`?
I saw that aliyun did such thing https://help.aliyun.com/document_detail/440233.html

And a background, CDH does not ship STS, so there is no conflict between hive beeline and spark beeline

cc @turboFei @yaooqinn

File: csd/kyuubi-csd-src/scripts/control.sh [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+  	KYUUBI_JAR_DIR="$KYUUBI_HOME/jars"
+    	if [[ ! -d ${KYUUBI_JAR_DIR} ]]; then
+  		echo -e "\nCandidate Kyuubi lib $KYUUBI_JAR_DIR doesn't exist, searching development environment..."
+    		KYUUBI_JAR_DIR="$KYUUBI_HOME/kyuubi-assembly/target/scala-${KYUUBI_SCALA_VERSION}/jars"
+  	fi
+    fi
+    if [[ -z ${YARN_CONF_DIR} ]]; then
+  	KYUUBI_CLASSPATH="${KYUUBI_JAR_DIR}/*:${KYUUBI_CONF_DIR}:${HADOOP_CONF_DIR}"
+    else
+  	KYUUBI_CLASSPATH="${KYUUBI_JAR_DIR}/*:${KYUUBI_CONF_DIR}:${HADOOP_CONF_DIR}:${YARN_CONF_DIR}"
```
Comment (pan3793) [inline csd/kyuubi-csd-src/scripts/control.sh]: the whole scripts is kind of hacky, `bin/kyuubi run` can run KyuubiServer in foreground, have you tried this way?

Comment (wangmiao1002) [reply csd/kyuubi-csd-src/scripts/control.sh]: No, at that time, the test used bin/kyuubi run to exit the process directly, and other processes were executing, for fear that the supervisor would not work, so this method was not used; Let me modify the test and use bin/kyuubi run

Comment (wangmiao1002) [reply csd/kyuubi-csd-src/scripts/control.sh]: There are still some problems with using bin/kyuubi script, kyuubi_ HOME, kyuubi-env.sh and other parameters in the script are set. cm is automatically generated according to the page options, especially the configuration file under conf. If the conf generated by cm is placed under kyuubi/conf in parcels, there are still some permission problems to debug. Please see if you need to do this

File: csd/kyuubi-csd-src/descriptor/service.sdl [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+  "gateway" : {
+    "alternatives" : {
+      "name" : "conf",
+      "priority" : 51,
+      "linkRoot" : "/etc/kyuubi"
+    },
+    "configWriter" : {
+       "peerConfigGenerators" : [
+          {
+            "filename": "conf/server.properties",
```
Comment (pan3793) [inline csd/kyuubi-csd-src/descriptor/service.sdl:150]: what's this?

File: csd/kyuubi-csd-src/scripts/control.sh [hunk 5]
```diff
@@ -0,0 +1,134 @@
+#!/bin/bash
+
+set -ex
+
+function log {
+  timestamp=$(date)
+  echo "$timestamp: $1"       #stdout
+  echo "$timestamp: $1" 1>&2; #stderr
```
Comment (pan3793) [inline csd/kyuubi-csd-src/scripts/control.sh]: why log to both stdout and stderr?

Comment (wangmiao1002) [reply csd/kyuubi-csd-src/scripts/control.sh]: Cm will generate stdout and stderr under each process. After the script is adjusted, the output of kyuubi will be directly used for printing. Can this output be removed

File: csd/kyuubi-csd-src/descriptor/service.sdl (hunk 4 above)
Comment (wangmiao1002) [reply csd/kyuubi-csd-src/descriptor/service.sdl:150]: It is used to generate client configuration, such as hive's hive-site.xml, which is saved on the gateway machine. The main reason for adding here is that there will be corresponding prompts to modify the configuration on the cm page. If you delete it, you will not be prompted. If you have any good suggestions, save some information in it

File: csd/kyuubi-csd-src/scripts/control.sh [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    echo appender.DRFA.policies.size.type=SizeBasedTriggeringPolicy >> conf/log4j2.properties
+    echo appender.DRFA.policies.size.size=\${max.log.file.size} >> conf/log4j2.properties
+    echo appender.DRFA.strategy.type=DefaultRolloverStrategy >> conf/log4j2.properties
+    echo appender.DRFA.strategy.max=\${max.log.file.backup.index} >> conf/log4j2.properties
+    echo appender.console.type=Console >> conf/log4j2.properties
+    echo appender.console.name=console >> conf/log4j2.properties
+    echo appender.console.target=SYSTEM_ERR >> conf/log4j2.properties
+    echo appender.console.layout.type=PatternLayout >> conf/log4j2.properties
+    echo appender.console.layout.pattern=%d{yy/MM/dd HH:mm:ss} %-5p %c{2}: [%t]: %m%n >> conf/log4j2.properties
+    exec ${KYUUBI_HOME}/bin/kyuubi run
```
Comment (pan3793) [inline csd/kyuubi-csd-src/scripts/control.sh:66]: https://github.com/apache/kyuubi/pull/4494 should make it work

File: csd/kyuubi-csd-src/descriptor/service.sdl [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+      "label": "Kyuubiserver",
+      "pluralLabel": "Kyuubi Server",
+      "jvmBased": true,
+      "startRunner": {
+        "program": "scripts/control.sh",
+        "args": [
+          "start"
+        ],
+      "environmentVariables":{
+                    "KYUUBI_WORK_DIR":"${kyuubi.session.work.path}",
```
Comment (pan3793) [inline csd/kyuubi-csd-src/descriptor/service.sdl]:
```suggestion
                    "KYUUBI_WORK_DIR_ROOT":"${kyuubi_work_dir_root}",
```

File: csd/kyuubi-csd-src/descriptor/service.sdl [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+		  "default": "USER",
+		  "type": "string_enum",
+		  "validValues": ["CONNECTION","USER","GROUP","SERVER"]
+          
+        },
+        {
+          "name": "kyuubi.session.work.path",
+          "label": "kyuubi.session.work.path",
+          "configName": "kyuubi.session.work.path",
+          "description": "Engines will be shared in different levels",
```
Comment (pan3793) [inline csd/kyuubi-csd-src/descriptor/service.sdl]:
```suggestion
          "name": "kyuubi_work_dir_root",
          "label": "kyuubi_work_dir_root",
          "configName": "kyuubi_work_dir_root",
          "description": "Root directory for launching sql engine applications.",
```

File: csd/kyuubi-csd-src/descriptor/service.sdl [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+          "configName": "kyuubi_work_dir_root",
+          "description": "Root directory for launching sql engine applications.",
+          "required": "true",
+		  "type": "path",
+		  "pathType": "localDataDir",
+          "default": "/data/kyuubi/work"
+        },
+		{
+          "name": "kyuubi.session.engine.idle.timeout",
+          "label": "Doris Frontend query Port",
```
Comment (pan3793) [inline csd/kyuubi-csd-src/descriptor/service.sdl]:
```suggestion
          "label": "kyuubi.session.engine.idle.timeout",
```

File: parcel/kyuubi-parcel-src/meta/alternatives.json [hunk 10]
```diff
@@ -0,0 +1,8 @@
+{
+    "kyuubi-beeline": {
+      "destination": "/usr/bin/kyuubi-beeline",
+      "source": "bin/beeline",
```
Comment (pan3793) [inline parcel/kyuubi-parcel-src/meta/alternatives.json:4]: this soft link does not work, will fix later

**Comment by pan3793** [comment]
@wangmiao1002 thanks for making this great job! I think it's fine to merge first and do other improvements in follow ups.

**Pull request merged**